aesd-char-driver/main.c

```diff
@@ -178,13 +197,147 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
     return retval;
 }
 
+/**
+ * @brief Repositions the file offset within the AESD device file.
+ *
+ * This function allows repositioning the file offset within the AESD device file
+ * associated with the given file pointer `filp`. It uses the provided `offset` and
+ * `whence` parameters to determine the new file offset.
+ *
+ * @param filp    Pointer to the file structure representing the opened AESD device file.
+ * @param off     The new file offset to set, which may be relative to the `whence` parameter.
+ * @param whence  The reference point for repositioning the file offset, such as SEEK_SET, SEEK_CUR, or SEEK_END.
+ *
+ * @return The new file offset after repositioning, or an error code if repositioning fails.
+ */
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+    struct aesd_dev *my_dev = filp->private_data;
+    loff_t return_offset;
+    /*
+    * Total size of the circular buffer is passed as last parameter to the below function
+    */
+    return_offset =  fixed_size_llseek(filp, offset, whence, my_dev->cir_buff_total_size);
```

Comment:
@Rishikesh38 if you don't lock the access to `my_dev->cir_buff_total_size` what does this assume about the `cir_buff_total_size` variable?